modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PlacementDriver.java

```diff
@@ -0,0 +1,108 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.replicator;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.ignite.internal.replicator.ReplicaService;
+import org.apache.ignite.internal.tx.TxMeta;
+import org.apache.ignite.internal.tx.message.TxStateReplicaRequest;
+import org.apache.ignite.lang.IgniteBiTuple;
+import org.apache.ignite.network.ClusterNode;
+
+/**
+ * Placement driver.
+ */
+public class PlacementDriver {
+    /** Assignment nodes per replication group. */
+    private final Map<String, LinkedHashSet<ClusterNode>> primaryReplicaMapping = new ConcurrentHashMap<>();
+
+    /** Replication service. */
+    private final ReplicaService replicaService;
+
+    /**
+     * The constructor.
+     *
+     * @param replicaService Replication service.
+     */
+    public PlacementDriver(ReplicaService replicaService) {
+        this.replicaService = replicaService;
+    }
+
+    /**
+     * Sent a transaction sate request to the primary replica.
+     *
+     * @param replicaGrp Replication group id.
+     * @param request Status request.
+     * @return Result future.
+     */
+    public CompletableFuture<TxMeta> sentMetaRequest(String replicaGrp, TxStateReplicaRequest request) {
+        CompletableFuture<TxMeta> resFut = new CompletableFuture<>();
+
+        sentAndRetry(resFut, replicaGrp, request);
+
+        return resFut;
+    }
+
+    /**
+     * Updates an assignment for the specific replication group.
+     *
+     * @param replicaGrpId Replication group id.
+     * @param assignment Assignment.
+     */
+    public void updateAssignment(String replicaGrpId, Collection<ClusterNode> assignment) {
+        primaryReplicaMapping.put(replicaGrpId, new LinkedHashSet<>(assignment));
+    }
+
+    /**
+     * Sent a request to primary replica of the replication group.
+     * If the primary node is changed the request is sending to a new primary node.
+     *
+     * @param resFut Response future.
+     * @param replicaGrp Replication group id.
+     * @param request Request.
+     */
+    private void sentAndRetry(CompletableFuture<TxMeta> resFut, String replicaGrp, TxStateReplicaRequest request) {
+        ClusterNode nodeToSent = primaryReplicaMapping.get(replicaGrp).iterator().next();
+
+        //TODO: Handle timout here. If an invocation is timed out the request should be passed to a next node from the assignment.
```

Comment:
Need to create ticket.

Reply:
I removed the comment, because our transaction implementation is designed for stable topology only.